ssl/tls13_enc.c

```diff
@@ -764,12 +772,132 @@ int tls13_change_cipher_state(SSL *s, int which)
         s->statem.enc_write_state = ENC_WRITE_STATE_WRITE_PLAIN_ALERTS;
     else
         s->statem.enc_write_state = ENC_WRITE_STATE_VALID;
+#ifndef OPENSSL_NO_KTLS
+# if defined(OPENSSL_KTLS_TLS13)
+#  ifndef __FreeBSD__
+    if (!(which & SSL3_CC_WRITE)
+        || ((which & SSL3_CC_WRITE) && (s->mode & SSL_MODE_NO_KTLS_TX)))
+        goto skip_ktls;
+
+    /* ktls supports only the maximum fragment size */
+    if (ssl_get_max_send_fragment(s) != SSL3_RT_MAX_PLAIN_LENGTH)
+        goto skip_ktls;
+
+    /* check whether cipher is known */
+    if (cipher == NULL)
+        goto skip_ktls;
```

Comment:
`cipher` shouldn't ever be NULL here I think. We already de-ref'd it in the `derive_secret_key_and_iv` call.

Reply:
Should this check be moved earlier, just before `derive_secret_key_and_iv` or we assume that NULL value of `cipher` is only possible in error path?

Reply:
Either we just remove it, or we could add this just before `derive_secret_key_and_iv`:

````
    if(!ossl_assert(cipher != NULL))
        goto err;
````

Reply:
Made this check as you suggested